crates/factor-key-value/src/lib.rs

```diff
@@ -19,11 +20,14 @@ use spin_key_value::{
 };
 pub use store::MakeKeyValueStore;
 
+/// A factor that provides key-value storage.
 pub struct KeyValueFactor {
+    /// Resolves runtime configuration into store managers.
     runtime_config_resolver: Arc<dyn runtime_config::RuntimeConfigResolver>,
 }
 
 impl KeyValueFactor {
+    /// Create a new KeyValueFactor.
```

Comment:
I think the important thing to document here is the `runtime_config_resolver` argument. Something like:

```suggestion
    /// Create a new KeyValueFactor.
    ///
    /// The `runtime_config_resolver` is used to resolve runtime configuration into store managers.
```